src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java

```diff
@@ -162,34 +162,47 @@ public void configure() {
                 .id("unzip-file")
                 .choice()
                     .when(isZippedFile("zip"))
-                        .split(new ZipSplitter())
+                        .split(new ZipSplitter()).aggregationStrategy(this::calculateICSAggregated)
                         .streaming()
                         .to("direct:calculate")
                     .endChoice()
                     .when(isZippedFile("tar.gz"))
                         .unmarshal().gzip()
-                        .split(new TarSplitter())
-                        .streaming()
-                        .to("direct:calculate")
+                        .split(new TarSplitter()).aggregationStrategy(this::calculateICSAggregated)
+                            .streaming()
+                            .to("direct:calculate")
+                        .end()
                     .endChoice()
                     .otherwise()
                         .to("direct:calculate")
-                .end();
+                        .process(exchange -> exchange.getIn().getHeaders().put("uploadformdata", exchange.getIn().getBody(UploadFormInputDataModel.class)))
+                    .end()
+                .end()
+                // at this time it's safe to generate the WSR
+                .to("direct:send-costsavings")
+                .to("direct:calculate-workloadsummaryreportmodel");
 
         from("direct:calculate")
                 .id("calculate")
                 .convertBodyTo(String.class)
-                .multicast()
+                .multicast().aggregationStrategy(new org.apache.camel.processor.aggregate.GroupedBodyAggregationStrategy())
```

Comment:
please remove the FQN and use an `import`

Reply:
+1